yogstation/code/modules/power/nuclear_engine.dm

```diff
@@ -0,0 +1,449 @@
+	var/SavedWarningTime = 0 //For cooldowns
+	var/WarningSound = 'yogstation/sound/effects/CoreOverheating.ogg'
+	var/MeltDownSound = 'yogstation/sound/effects/Meltdown.ogg'
+	var/ReactorInoperable = FALSE
+	density = TRUE
+//	var/CoreHealth = 1000 //This gets chipped away when you run it above its temperature tolerance
+
+/obj/machinery/power/NuclearReactor/examine(mob/user)
+	. = ..()
+	to_chat(user, "Its heat readout reads: [Heat] K, Its maximum tolerance limit is [CRITICAL_HEAT] K")
+
+
+//COOL STUFF: PIPES AIR CONTAINER: AIRS | AIR TEMPERATURE VAR: TEMPERATURE (K) aim for 1000 for turbines!
+//var/newrpm = ((compressor.gas_contained.temperature) * compressor.gas_contained.total_moles())/4
+
+
+
+/obj/effect/reactoroverlay
+	name = "reactor component"
+	desc = "why are you seeing this"
+	icon = 'icons/obj/reactor.dmi'
+	icon_state = "hatch-closed"
+
+/obj/effect/reactoroverlay/fuelhatch
+	icon_state = "hatch-closed"
+
+/obj/effect/reactoroverlay/wastehatch
+	icon_state = "wastehatch-closed"
+
+/obj/effect/reactoroverlay/monitor
+	icon_state = "monitor-off"
+
+/obj/machinery/power/NuclearReactor/CtrlClick(mob/user)
+	if(state & FUELHATCH_OPEN)
+		state &= ~FUELHATCH_OPEN //Bitflags! We add the OPPOSITE of fuelhatch open to "state", in other words, inverting it, if the hatch is open, this closes it
+		playsound(loc, 'sound/effects/bin_close.ogg',50,1)
+		to_chat(user,"You shut the fuel hatch")
+	else
+		state |= FUELHATCH_OPEN
+		playsound(loc, 'sound/effects/bin_open.ogg',50,1)
+		to_chat(user,"You open the fuel hatch")
+	UpdateIcon()
+
+/obj/machinery/power/NuclearReactor/AltClick(mob/user)
+	if(state & WASTEHATCH_OPEN)
+		state &= ~WASTEHATCH_OPEN //Bitflags! We add the OPPOSITE of wastehatch open to "state", in other words, inverting it, if the hatch is open, this closes it
+		playsound(loc, 'sound/effects/bin_close.ogg',50,1)
+		to_chat(user,"You shut the access hatch")
+	else
+		state |= WASTEHATCH_OPEN
+		playsound(loc, 'sound/effects/bin_open.ogg',50,1)
+		to_chat(user,"You open the access hatch")
+	UpdateIcon()
+
+/obj/machinery/power/NuclearReactor/proc/UpdateIcon()
+	cut_overlays()
+	waste.icon_state = "wastehatch-closed"
+	fuel.icon_state = "hatch-closed"
+	if(state & FUELHATCH_OPEN)
+		fuel.icon_state = "hatch-open"
+	if(state & WASTEHATCH_OPEN)
+		waste.icon_state = "wastehatch-open"
+	if(Heat >= LOW_HEAT)
+		monitor.icon_state = "monitor-low"
+	if(Heat > MEDIUM_HEAT)
+		monitor.icon_state = "monitor-med"
+	if(Heat > HIGH_HEAT)
+		monitor.icon_state = "monitor-high"
+	if(Heat > CRITICAL_HEAT) //At this stage. Worry. A lot.
+		monitor.icon_state = "monitor-warning"
+	add_overlay(waste)
+	add_overlay(fuel)
+	add_overlay(monitor)
+
+/obj/machinery/power/NuclearReactor/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/twohanded/required/FuelRod))
+		if(state & FUELHATCH_OPEN)
+			to_chat(user, "You slot [I] into [src]!")
+			I.forceMove(src)
+			react.FRS += I
+		else
+			to_chat(user, "Open the fuel hatch first!")
```

Comment:
How about some `<span class='warning'>` here and there? Never hurts to make things colorful